examples/x-and-xx.007

```diff
@@ -0,0 +1,39 @@
+macro infix:<xx>(left, right) is equal(infix:<*>) {
+    # Flattens one layer.
+    sub flatten (inputList) {
+        my result = [];
+        for inputList -> i {
+            if (type(i) == type([])) {
+                for i -> j {
+                    result.push(j);
+                }
```

Comment:
Shorter and no less clear than the `for` loop might be to do `result = result.concat(i);` &mdash; this appends all the elements "at once" at the expense of creating a brand new array and letting the old one be collected.